src/v/cluster/controller_probe.cc

```diff
@@ -101,6 +102,25 @@ void controller_probe::setup_metrics() {
           sm::description(
             "Number of partitions that lack quorum among replicants"))
           .aggregate({sm::shard_label}),
+        sm::make_gauge(
+          "latest_cluster_metadata_manifest_age",
+          [this] {
+              auto maybe_manifest_ref
+                = _controller.metadata_uploader().manifest();
+              if (!maybe_manifest_ref.has_value()) {
+                  return int64_t{-1};
+              }
+
+              auto const& manifest = maybe_manifest_ref.value().get();
+              auto now_ts = ss::lowres_system_clock::now().time_since_epoch();
+
+              auto age_s = std::chrono::duration_cast<std::chrono::seconds>(
+                now_ts - manifest.upload_time_since_epoch);
+              return int64_t{age_s.count()};
```

Comment:
If you're looking for a place to test this, I think cluster/cloud_metadata/tests/uploader_test.cc would be a good place. The full application is in that test, so we _should_ have metrics.